File: src/models/ProjectMilestone.ts
import type { MilestoneCreatorRole, MilestoneStatus } from "@/types/auth";

export type ProjectMilestoneRow = {
  id: string;
  project_id: string;
  title: string;
  description: string | null;
  status: MilestoneStatus;
  attachment_urls: string[];
  due_date: string | null;
  completed_at: string | null;
  created_by_role: MilestoneCreatorRole;
  created_by_id: string;
  created_at: string;
  updated_at: string;
};

export type ProjectMilestone = {
  id: string;
  projectId: string;
  title: string;
  description: string | null;
  status: MilestoneStatus;
  attachmentUrls: string[];
  dueDate: string | null;
  completedAt: string | null;
  createdByRole: MilestoneCreatorRole;
  createdById: string;
  createdAt: string;
  updatedAt: string;
};

File: rkispro-android/app/src/main/java/com/rkispro/marketplace/data/model/ProjectMilestone.kt
package com.rkispro.marketplace.data.model

import com.google.gson.annotations.SerializedName

data class ProjectMilestone(
    val id: String,
    @SerializedName("project_id") val projectId: String,
    val title: String,
    val description: String? = null,
    val status: String = "pending",
    @SerializedName("due_date") val dueDate: String? = null,
    @SerializedName("completed_at") val completedAt: String? = null,
    @SerializedName("created_by_role") val createdByRole: String,
    @SerializedName("created_by_id") val createdById: String,
    @SerializedName("created_at") val createdAt: String? = null,
    @SerializedName("updated_at") val updatedAt: String? = null
)
